src/ol/interaction/Draw.js

```diff
@@ -632,9 +632,10 @@ class Draw extends PointerInteraction {
               break;
             }
           // eslint-disable-next-line no-fallthrough
-          case this.freehand_ ||
-            (this.atFinish_(event.pixel) && this.finishCondition_(event)):
-            this.finishDrawing();
+          case this.freehand_ || this.atFinish_(event.pixel):
+            if (this.freehand_ || this.finishCondition_(event)) {
+              this.finishDrawing();
+            }
```

Comment:
When reading this change, it is not obvious at first glance why the previous version did not work. Instead of falling through from the previous `case`, wouldn't it be possible to get rid of this `case`? I have not tested the suggestion below, but I think something like this could work:
```suggestion
            if (this.freehand_ ||
             (this.atFinish_(event.pixel) && this.finishCondition_(event))) {
              this.finishDrawing();
            }
```